langchain-core/src/vectorstores.ts

```diff
@@ -337,6 +769,41 @@ export abstract class VectorStore
     );
   }
 
+  /**
+   * Creates a `VectorStoreRetriever` instance with flexible configuration options.
+   *
+   * @param kOrFields
+   *    - If a number is provided, it sets the `k` parameter (number of items to retrieve).
+   *    - If an object is provided, it should contain various configuration options.
+   * @param filter
+   *    - Optional filter criteria to limit the items retrieved based on the specified filter type.
+   * @param callbacks
+   *    - Optional callbacks that may be triggered at specific stages of the retrieval process.
+   * @param tags
+   *    - Tags to categorize or label the `VectorStoreRetriever`. Defaults to an empty array if not provided.
+   * @param metadata
+   *    - Additional metadata as key-value pairs to add contextual information for the retrieval process.
+   * @param verbose
+   *    - If `true`, enables detailed logging for the retrieval process. Defaults to `false`.
+   *
+   * @returns
+   *    - A configured `VectorStoreRetriever` instance based on the provided parameters.
+   *
+   * @example
+   * // Basic usage with a `k` value:
+   * const retriever = asRetriever(5);
+   *
+   * @example
+   * // Usage with a configuration object:
+   * const retriever = asRetriever({
```

Comment:
Example is wrong - this is an instance method

Also should wrap in markdown blocks for syntax highlighting in API refs